lib/tlIO/FFmpegWrite.cpp

```diff
@@ -42,198 +176,582 @@ namespace tl
+                    ss >> timecode;
+                    r = av_dict_set(
+                        &p.avFormatContext->metadata, "timecode", timecode.c_str(), 0);
+                    if (r < 0)
+                        throw std::runtime_error(string::Format("Could not set timecode to {1}")
+                                                 .arg(timecode));
+                
+                    otime::ErrorStatus errorStatus;
+                    const otime::RationalTime time = otime::RationalTime::from_timecode(
+                        timecode,
+                        info.videoTime.duration().rate(),
+                        &errorStatus);
+                    if (!otime::is_error(errorStatus))
+                    {
+                        p.videoStartTime = time::floor(time);
+                    }
+                }
+
+                p.avPacket = av_packet_alloc();
+                if (!p.avPacket)
+                {
+                    throw std::runtime_error(string::Format("{0}: Cannot allocate packet").arg(p.fileName));
+                }
+
+                p.avFrame = av_frame_alloc();
+                if (!p.avFrame)
+                {
+                    throw std::runtime_error(string::Format("{0}: Cannot allocate frame").arg(p.fileName));
+                }
+                p.avFrame->format = p.avVideoStream->codecpar->format;
+                p.avFrame->width = p.avVideoStream->codecpar->width;
+                p.avFrame->height = p.avVideoStream->codecpar->height;
+                r = av_frame_get_buffer(p.avFrame, 0);
+                if (r < 0)
+                {
+                    throw std::runtime_error(
+                        string::Format("{0}: av_frame_get_buffer - {1}")
+                            .arg(p.fileName)
+                            .arg(getErrorLabel(r)));
+                }
+
+                p.avFrame2 = av_frame_alloc();
+                if (!p.avFrame2)
+                {
+                    throw std::runtime_error(string::Format("{0}: Cannot allocate frame").arg(p.fileName));
+                }
+                switch (videoInfo.pixelType)
+                {
+                case image::PixelType::L_U8:     p.avPixelFormatIn = AV_PIX_FMT_GRAY8;  break;
+                case image::PixelType::RGB_U8:   p.avPixelFormatIn = AV_PIX_FMT_RGB24;  break;
+                case image::PixelType::RGBA_U8:  p.avPixelFormatIn = AV_PIX_FMT_RGBA;   break;
+                case image::PixelType::L_U16:    p.avPixelFormatIn = AV_PIX_FMT_GRAY16; break;
+                case image::PixelType::RGB_U16:  p.avPixelFormatIn = AV_PIX_FMT_RGB48;  break;
+                case image::PixelType::RGBA_U16: p.avPixelFormatIn = AV_PIX_FMT_RGBA64; break;
+                default:
+                    throw std::runtime_error(string::Format("{0}: Incompatible pixel type").arg(p.fileName));
+                    break;
+                }
+                /*p.swsContext = sws_getContext(
```

Comment:
This commented out code could be removed, I only kept it temporarily to make sure the `sws_alloc_context()` worked as a replacement.